File: ui/Transport.py
__author__ = 'toramisu'

from ui import *
from module.Events import *


class Transport(QWidget):
    def __init__(self, parent):
        super(Transport, self).__init__(parent)
        self.hbox = QHBoxLayout(self)
        self.hbox.setSpacing(0)
        self.frameRate = QComboBox()
        self.frameRate.addItems(['24', '30', '60'])
        connect(self.frameRate.currentTextChanged, self.onFrameRateChanged)
        self.hbox.addWidget(self.frameRate)

        self.playButton = QPushButton()
        self.playButton.setText('play')
        Event.add(ActionEvent.TOGGLE_PLAY, self.togglePlay)
        connect(self.playButton.clicked, self.togglePlay)
        self.hbox.addWidget(self.playButton)

        self.pauseButton = QPushButton()
        self.pauseButton.setText('stop')
        self.hbox.addWidget(self.pauseButton)
        # self.setStyleSheet('QWidget { background: red; }')
        pass

    def onFrameRateChanged(self, e):
        from module import App

        fr = int(self.frameRate.currentText())
        App().sequencePlayback.setFramerate(fr)
        pass

    def togglePlay(self):
        from module import App
        state = App().sequencePlayback.state
        if state == PlayStateType.PLAY:
            self.playButton.setText('play')
            Event.dis(PlaybackEvent.STATE, PlayStateType.PAUSE)
        else:
            self.playButton.setText('pause')
            Event.dis(PlaybackEvent.STATE, PlayStateType.PLAY)
        pass

File: view/viewport/Transport.cpp
//
// Created by manoyuria on 2015/6/19.
//

#include "Transport.h"
Transport::Transport(QWidget *parent) : QWidget(parent) {
    QHBoxLayout *hbox = new QHBoxLayout(this);
    hbox->setSpacing(0);

    QComboBox *frameRate = new QComboBox();
    frameRate->addItems({"24", "30", "60"});
    hbox->addWidget(frameRate);
    connect(frameRate, QComboBox::currentTextChanged, this, onFrameRateChanged);
//
    QPushButton *playButton = new QPushButton();
    playButton->setText("play");
//    Evt_add(ActionEvent_TOGGLE_PLAY,togglePlay);
    connect(playButton,QPushButton::clicked,this,togglePlay);
    hbox->addWidget(playButton);

    QPushButton *stopButton = new QPushButton();
    stopButton->setText("stop");
    hbox->addWidget(stopButton);
}
void Transport::onFrameRateChanged()
{

}

void Transport::togglePlay() {

}
